pkg/build/webhook/generic/generic.go

```diff
@@ -44,7 +45,10 @@ func (p *WebHookPlugin) Extract(buildCfg *api.BuildConfig, secret, path string,
 		return nil, true, nil
 	}
 
-	if req.Body != nil && req.Header.Get("Content-Type") == "application/json" {
+	contentType := req.Header.Get("Content-Type")
+	mediaType, _, err := mime.ParseMediaType(contentType)
```

Comment:
if it's actually unparseable we should probably still return/report the error (because it implies someone was trying to supply a value, so we don't want to silently ignore it).  if it's empty/not supplied, I assume that won't result in an error, will it?